core/query.go

```diff
@@ -46,27 +54,36 @@ type Query struct {
 	// The DagQL query cache.
 	Cache dagql.Cache
 
+	BuildkitOpts *buildkit.Opts
+	Recorder     *progrock.Recorder
+
 	// The metadata of client calls.
 	// For the special case of the main client caller, the key is just empty string.
 	// This is never explicitly deleted from; instead it will just be garbage collected
 	// when this server for the session shuts down
-	clientCallContext map[digest.Digest]*ClientCallContext
-	clientCallMu      *sync.RWMutex
+	ClientCallContext map[digest.Digest]*ClientCallContext
+	ClientCallMu      *sync.RWMutex
 
 	// the http endpoints being served (as a map since APIs like shellEndpoint can add more)
-	endpoints  map[string]http.Handler
-	endpointMu *sync.RWMutex
+	Endpoints  map[string]http.Handler
+	EndpointMu *sync.RWMutex
 }
 
-var ErrNoCurrentModule = fmt.Errorf("no current module")
+func NewRoot(ctx context.Context, opts QueryOpts) (*Query, error) {
+	// TODO: double check context won't result in something getting cancelled prematurely
```

Comment:
reminder to self to trace this carefully again (though tests have been passing so I think it's fine)

Reply:
Confirmed this should be good; did some cleanup to use `WithoutCancel` too in this commit https://github.com/dagger/dagger/pull/6806/commits/e7e7235556bec84308c635007d12d3dfc78e577e